tests/geomodel/core/test-geomodel-build.cpp

```diff
@@ -0,0 +1,641 @@
+    // Line 4 is a boundary of the surfaces: 1, 3 and 4.
+    builder.topology.add_surface_line_boundary_relation( 4, 4 );
+    builder.topology.add_surface_line_boundary_relation( 6, 4 );
+
+    // For line 5
+    // Line 5 is a boundary of the surfaces: 1, 9 and 10.
+    builder.topology.add_surface_line_boundary_relation( 4, 5 );
+    builder.topology.add_surface_line_boundary_relation( 5, 5 );
+
+    // For line 6
+    // Line 6 is a boundary of the surfaces: 1, 5 and 6.
+    builder.topology.add_surface_line_boundary_relation( 1, 6 );
+    builder.topology.add_surface_line_boundary_relation( 2, 6 );
+
+    // For line 7
+    // Line 7 is a boundary of the surfaces: 1, 7 and 8.
+    builder.topology.add_surface_line_boundary_relation( 5, 7 );
+    builder.topology.add_surface_line_boundary_relation( 6, 7 );
+
+    // For line 8
+    // Line 8 is a boundary of the surfaces: 2 and 3.
+    builder.topology.add_surface_line_boundary_relation( 0, 8 );
+    builder.topology.add_surface_line_boundary_relation( 3, 8 );
+    builder.topology.add_surface_line_boundary_relation( 5, 8 );
+
+    /////////////////////////////////////////////////////////
+
+    // For surface 0
+    // Surface 0 is a boundary of the region 0.
+    builder.topology.add_region_surface_boundary_relation(
+        0, 0, false ); // TODO side ????
+
+    // For surface 1
+    // Surface 1 is a boundary of the region 0.
+    builder.topology.add_region_surface_boundary_relation(
+        0, 1, false ); // TODO side ????
+
+    // For surface 2
+    // Surface 2 is a boundary of the region 1.
+    builder.topology.add_region_surface_boundary_relation(
+        0, 2, false ); // TODO side ????
+
+    // For surface 3
+    // Surface 3 is a boundary of the region 1.
+    builder.topology.add_region_surface_boundary_relation(
+        0, 3, false ); // TODO side ????
+
+    // For surface 4
+    // Surface 4 is a boundary of the region 0.
+    builder.topology.add_region_surface_boundary_relation(
+        1, 4, false ); // TODO side ????
+
+    // For surface 5
+    // Surface 5 is a boundary of the region 1.
+    builder.topology.add_region_surface_boundary_relation(
+        1, 5, false ); // TODO side ????
+
+    // For surface 6
+    // Surface 6 is a boundary of the region 0.
+    builder.topology.add_region_surface_boundary_relation(
+        1, 6, false ); // TODO side ????
+
+    //#####################################
+    // Setting the parent/child relations #
+    //#####################################
+
+    // Remember :
+    // Child of a Contact is a Line
+    // Child of an Interface is a Surface
+    // Child of a Layer is a Region
+
+    // We use the method "add_parent_children_relation"
+    // First argument is the parent (ie a GeoModelGeologicalEntity)
+    // Second argument is the index of the child (ie a GeoModelMeshEntity)
+
+    // For Contact 0
+    builder.geology.add_parent_children_relation(
+        gmge_id( Contact3D::type_name_static(), 0 ),
```

Comment:
you may use "{ Contact3D::type_name_static(), 0}" instead of "gmge_id( Contact3D::type_name_static(), 0 ) "(may need #include <ringmesh/mesh/mesh_index.h>)